.github/workflows/ci.yml

```diff
@@ -130,7 +170,14 @@ jobs:
           echo "disk space is:"
           df -h
 
-      - name: C++ Unit Tests
+      - name: C++ Unit Tests sanitizers
+        if: matrix.sanitizers == 'Sanitizers'
+        run: |
+            cd ${GITHUB_WORKSPACE}/build
+            ctest -V -L DFLY
```

Comment:
what is the reason for different unit test run for sanitizers and the "plain"?

Reply:
We run unit tests with epoll and without and also some more other manually :shrug:

Reply:
you are stating what we are doing. I can see that and therefor I ask what is the reason for this? 
Why not run the same for both?

Reply:
Note that if you unite the step and run the same tests for both you will not need the matrix.sanitizers at all

Reply:
Helio is a third party lib, not part of dragonfly. We are using sanitizers to test dragonfly and not helio's proactor (iouring or epoll). Furthermore, you are trying to to bring everything under the same umbrella. I don't think this is a good thing. The end goal would be to have separate actions, one for `normal builds and one for sanitizers`. That way, as we change things, workflows are separate and self contained (similar to what we do with regression tests, that we have action which we parametarize). 

Anyway, these are not important for now. I will unify them for now

Reply:
I believe that unifying the steps fro different builds (similarly to code quadruplication )
1. removes rows and simplifies 
2. if we need to do changes we dont need to remember to fix both

Unlsess there is a reason to not unify them I dont see why not.
We want the sanitizers to run on different flows, there is a reason we have in the build running tests with different flags for example cluster emulated / lock on hash tag. If we run the sanitizers on this flows as well we might catch different failures.

Similar to unifying the bulild step, you see that we added some params to the build f.e HELIO_STACK_CHECK that we forgot to add to the sanitizers bulid. once we unify this there is nothing to remember